pydantic/fields.py

```diff
@@ -394,68 +395,68 @@ class MyModel(pydantic.BaseModel):
                 code='unevaluable-type-annotation',
             )
 
-        final = _typing_extra.is_finalvar(annotation)
-        if final and _typing_extra.is_generic_alias(annotation):
-            annotation = typing_extensions.get_args(annotation)[0]
+        try:
+            inspected_ann = inspect_annotation(
+                annotation,
+                annotation_source=_source,
+                unpack_type_aliases='skip',
+            )
+        except ForbiddenQualifier as e:
+            raise PydanticForbiddenQualifier(e.qualifier, annotation)
 
-        if isinstance(default, FieldInfo):
-            if _typing_extra.is_annotated(annotation):
-                annotation, *annotation_metadata = typing_extensions.get_args(annotation)
-            else:
-                annotation_metadata = []
+        # TODO check for classvar and error?
 
-            default.annotation = annotation
-            default.metadata += annotation_metadata
-            default = default.merge_field_infos(
-                *[x for x in annotation_metadata if isinstance(x, FieldInfo)], default, annotation=default.annotation
+        # TODO infer from the default
```

Comment:
This can wait for v3, as if the field has a default it is currently treated as a class var anyway. This todo is thus linked in https://github.com/pydantic/pydantic/issues/11119.